modin/core/dataframe/algebra/binary.py

```diff
@@ -96,13 +160,23 @@ def caller(
                         )
                     )
                 else:
+                    if other.dtypes is not None and query_compiler.dtypes is not None:
```

Comment:
```suggestion
                    if other._modin_frame._dtypes is not None and query_compiler._modin_frame._dtypes is not None:
```

Reply:
Please also fix other places where .dtypes is accessed.

Reply:
Changed .dtypes to  ._modin_frame._dtypes to be aligned with the code.